code/modules/projectiles/guns/pistols.dm

```diff
@@ -75,6 +75,47 @@
 	recoil = -2
 	recoil_unwielded = -2
 
+//-------------------------------------------------------
+//TX-7 Plasma Pistol
+/obj/item/weapon/gun/pistol/plasma_pistol
+	name = "\improper TX-7 plasma pistol"
+	desc = "An experimental weapon. Designed to set the terrain it is fired at on fire so as to remove flammable obstacles and clear out dug in enemies from some range."
+	icon_state = "tx7"
+	item_state = "tx7"
+	caliber = "Ionized Plasma Bolt"
+	max_shots = 10
+	reload_sound = 'sound/weapons/guns/interact/tp14_reload.ogg'
+	fire_sound = 'sound/weapons/guns/fire/laser.ogg'
```

Comment:
You should check the other sounds of the laser gun, some would feel like a plasma gun more IMO